### PR TITLE
tenant_load_gen_script

### PR DESCRIPTION
**Pull request by yosharma**
script to create a tenant and generate load for the tenant. The script also cleans the created load for the tenant. The script also provides the option to create specific resource for a tenant along with all resources.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.generate_load_real_env
devlab/tests/generate_load_real_env.py:3: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/generate_load_real_env.py:34: [W0621(redefined-outer-name), clean_if_exists] Redefining name 'func' from outer scope (line 707)
devlab/tests/generate_load_real_env.py:35: [W0621(redefined-outer-name), clean_if_exists.wrapper] Redefining name 'args' from outer scope (line 691)
devlab/tests/generate_load_real_env.py:54: [W0621(redefined-outer-name), retry_until_resources_created.actual_decorator] Redefining name 'func' from outer scope (line 707)
devlab/tests/generate_load_real_env.py:56: [W0612(unused-variable), retry_until_resources_created.actual_decorator.wrapper] Unused variable 'i'
devlab/tests/generate_load_real_env.py:97: [W0703(broad-except), Prerequisites.create_keypairs] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:217: [E1101(no-member), Prerequisites.create_security_grp] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:307: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:316: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:537: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:546: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:556: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:564: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:572: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:587: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:601: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:611: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:623: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:638: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:652: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception
devlab/tests/generate_load_real_env.py:664: [W0703(broad-except), Prerequisites.clean_objects] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
- Please add description
- Please resolve all pylint issues

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.generate_load_real_env
devlab/tests/generate_load_real_env.py:3: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/generate_load_real_env.py:219: [E1101(no-member), Prerequisites.create_security_grp] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:309: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:318: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:548: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nv_exceptions
devlab/tests/generate_load_real_env.py:558: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: gl_exceptions
devlab/tests/generate_load_real_env.py:566: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:573: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:602: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:612: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:623: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: gl_exceptions
devlab/tests/generate_load_real_env.py:653: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: cd_exceptions
devlab/tests/generate_load_real_env.py:665: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: cd_exceptions

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.generate_load_real_env
devlab/tests/generate_load_real_env.py:3: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/generate_load_real_env.py:219: [E1101(no-member), Prerequisites.create_security_grp] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:309: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:318: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load_real_env.py:548: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nv_exceptions
devlab/tests/generate_load_real_env.py:558: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: gl_exceptions
devlab/tests/generate_load_real_env.py:566: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:573: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:602: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:612: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: nt_exceptions
devlab/tests/generate_load_real_env.py:623: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: gl_exceptions
devlab/tests/generate_load_real_env.py:653: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: cd_exceptions
devlab/tests/generate_load_real_env.py:665: [E0712(catching-non-exception), Prerequisites.clean_objects] Catching an exception which doesn't inherit from BaseException: cd_exceptions

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
please resolve all pylint issues

File: devlab/tests/generate_load_real_env.py [hunk 1]
```diff
@@ -17,83 +86,244 @@ def create_keypairs(self):
...
+            tenant['name'] = self.prefix + tenant['name']
+            for tnt in self.keystoneclient.tenants.list():
+                if tnt.name == tenant['name']:
+                    print ">>>Tenant", tenant['name'],
+                    print "and its resources already exist."
+                    print ">>>Please run --clean to clean tenant resources"
+                    logging.info(">>>Tenant %s & its resources already exist",
+                                 tenant['name'])
+                    logging.info(">>>Please run --clean first to"
+                                 " clean existing tenant resources")
```
Comment (roman-verchikov) [inline devlab/tests/generate_load_real_env.py]: mixture of `logging` and `print`. Please leave only one of the types of output, I'd suggest leaving `logging`.

File: devlab/tests/generate_load_real_env.py [hunk 2]
```diff
@@ -128,115 +377,341 @@ def emulate_vm_states(self):
...
+
+    def update_name_with_prefix(self, resource_name=None):
+        new_name = self.prefix + resource_name
+        return new_name
+
+    def create_options(self, function=None):
+        ''' This function is used to create resources
+            depending upon the option provided for the
+            tenant in configuration file '''
+        print "going to call :" + function
```
Comment (roman-verchikov) [inline devlab/tests/generate_load_real_env.py]: `print` again

File: devlab/tests/generate_load_real_env.py [hunk 3]
```diff
@@ -128,115 +377,341 @@ def emulate_vm_states(self):
...
-                self.cinderclient.volumes.delete(
-                    self.get_volume_id(volume['name']))
-            except Exception as e:
-                print "Volume %s failed to delete: %s" % (volume['name'],
-                                                          repr(e))
+                time.sleep(5)
+                self.cinderclient.volumes.delete(self.cinderclient.volumes.get(
+                    self.get_volume_id(volume['name'])))
+                logging.info('>>> Volume: : "%s" Deleted !!', volume['name'])
+            except (cd_exceptions, generate_load.NotFound):
```
Comment (roman-verchikov) [inline devlab/tests/generate_load_real_env.py]: Pattern
```
logging.info("removing {object_type}")
for object in objects:
    try:
        object.delete()
    except client.Exception:
         pass
```
is repeated many times, consider refactoring

**Comment by roman-verchikov** [comment]
This still requires resolution of pylint issues and few inlines

**Comment by roman-verchikov** [comment]
closed due to no activity

**Pull request closed without merging**